### PR TITLE
feat(cd): Add trivy docker-cis scan

### PR DESCRIPTION
**Pull request by 425devon**
Adds trivy docker-cis scan to Scan Docker Image shared action, tested with Scimia/shared-actions in my sandbox [HERE](https://github.com/Scimia/devons-sandbox/actions/runs/4147855003) and implemented in Scimia/shared-actions [HERE](https://github.com/Kong/shared-actions/compare/main...Scimia:shared-actions:feat/docker-cis-scanning)

### CODE REVIEW

File: security-actions/scan-docker-image/action.yml [hunk 1]
```diff
@@ -171,4 +174,20 @@ runs:
...
         fail-build: ${{ steps.meta.outputs.global_enforce_build_failure == 'true' && steps.meta.outputs.global_enforce_build_failure || inputs.fail_build }}
-        severity-cutoff: ${{ steps.meta.outputs.global_severity_cutoff }}
\ No newline at end of file
+        severity-cutoff: ${{ steps.meta.outputs.global_severity_cutoff }}
+
+    - name: Generate docker-cis report
+      shell: bash
+      id: cis_json
+      run: |
+        wget https://github.com/aquasecurity/trivy/releases/download/v0.37.1/trivy_0.37.1_Linux-64bit.deb
```
Comment (saisatishkarra) [inline security-actions/scan-docker-image/action.yml:183]: Can we make this silent and not pollute the console log with the progress?

Comment (425devon) [reply security-actions/scan-docker-image/action.yml]: Added the flag  "-q"  to turn off Wget's output. "-nv" non-verbose is another alternative...

File: security-actions/scan-docker-image/action.yml [hunk 2]
```diff
@@ -171,4 +174,20 @@ runs:
...
         fail-build: ${{ steps.meta.outputs.global_enforce_build_failure == 'true' && steps.meta.outputs.global_enforce_build_failure || inputs.fail_build }}
-        severity-cutoff: ${{ steps.meta.outputs.global_severity_cutoff }}
\ No newline at end of file
+        severity-cutoff: ${{ steps.meta.outputs.global_severity_cutoff }}
+
+    - name: Generate docker-cis report
+      shell: bash
+      id: cis_json
+      run: |
+        wget -q https://github.com/aquasecurity/trivy/releases/download/v0.37.1/trivy_0.37.1_Linux-64bit.deb
```
Comment (saisatishkarra) [inline security-actions/scan-docker-image/action.yml:183]: downloading the installer and pointing at a specific version and manually updating it everytime on breaking changes for trivy db / cache is difficult. Instead we should try using the  will help dependabot in the future to manage deps.

```
uses: docker://ghcr.io/aquasecurity/trivy:0.37.2
with:
    entrypoint: trivy
     rgs: "space separated command"
```

Also, based on https://github.com/orgs/community/discussions/25637, GitHub mounts job workspace inside the step container to share and access output files generated by the action to upload back as workflow artifacts.

File: security-actions/scan-docker-image/action.yml [hunk 3]
```diff
@@ -171,4 +174,20 @@ runs:
...
+        sudo dpkg -i trivy_0.37.1_Linux-64bit.deb
+        trivy image --input ${{ steps.meta.outputs.scan_image }} --compliance docker-cis -f table
```
Comment (saisatishkarra) [inline security-actions/scan-docker-image/action.yml:185]: Run this step after capturing the JSON file. All console log steps must be at the end of action for quick references

**Goal**: 
2nd Iteration with Table format:
Display severities in the console log honoring ignores / custom configuration (.trivyigniore in repo-root) but use either  `fail_build ` / `global_enforce_build_failure ` to set exit code for matched `global_severity_cutoff` when set.

Is this possible with two runs of trivy?

File: security-actions/scan-docker-image/action.yml [hunk 4]
```diff
@@ -171,4 +174,20 @@ runs:
...
+        trivy image --input ${{ steps.meta.outputs.scan_image }} --compliance docker-cis -f json > ${{ steps.meta.outputs.cis_json_file }}
```
Comment (saisatishkarra) [inline security-actions/scan-docker-image/action.yml:186]: 1st iteration of JSON output: Capture all severity matches from trivy including the ones ignored explicitly which allow the security team for in-depth analysis and app teams to document the reasoning for ignore and never fail based on cutoff / fail_build inputs

File: security-actions/scan-docker-image/action.yml [hunk 5]
```diff
@@ -171,4 +174,20 @@ runs:
...
+    - name: upload docker-cis report
```
Comment (saisatishkarra) [inline security-actions/scan-docker-image/action.yml:187]: Move immediately after generating JSON output and before table format step

**Review by saisatishkarra — CHANGES_REQUESTED** [review]

**Comment by saisatishkarra** [comment]
cc: @vs-kong  Devon is working to integrate docker cis benchmarks action using trivy alongside the container and vulnerability scanning. Please add any inputs/thoughts.

**Comment by saisatishkarra** [comment]
@425devon How does trivy manage what version `docker-cis` benchmarks are being used to evaluate the container image? Is there a way to control/manage this parameter?

**Comment by 425devon** [comment]
Trivy's Docker CIS Benchmark scanning has only been released for around a week now as of v0.37.0 Currently the only CIS benchmark supported is [CIS Docker Community Edition Benchmark v1.1.0](https://aquasecurity.github.io/trivy/v0.37/docs/target/container_image/) It does look like they will support multi platform builds however I've yet to test it. Here's a decent summary of the new features: https://www.youtube.com/watch?v=3T1uXos7BXU

File: security-actions/scan-docker-image/action.yml (hunk 4 above)
Comment (425devon) [reply security-actions/scan-docker-image/action.yml]: ✅

File: security-actions/scan-docker-image/action.yml (hunk 5 above)
Comment (425devon) [reply security-actions/scan-docker-image/action.yml]: ✅

File: security-actions/scan-docker-image/action.yml (hunk 3 above)
Comment (425devon) [reply security-actions/scan-docker-image/action.yml]: Addressed in latest push

File: security-actions/scan-docker-image/action.yml (hunk 2 above)
Comment (425devon) [reply security-actions/scan-docker-image/action.yml]: This was a great idea! Implemented.

**Comment by saisatishkarra** [comment]
@425devon 
- can you update dependabot.yml to use "docker" ecosystem and point to specific action directory instead of /.
- Please resolve the conflicts

**Review by saisatishkarra — APPROVED** [review]
lgtm

**Pull request merged**